contracts/p1/mixins/TradingLib.sol

```diff
@@ -43,18 +44,24 @@ library TradingLibP1 {
         trade.buy = buy;
 
         // Don't sell dust
-        if (!isEnoughToSell(sell, sellAmount, minTradeVolume())) return (false, trade);
+        if (!isEnoughToSell(sell, sellAmount, trader.minTradeVolume())) return (false, trade);
 
         // {sellTok}
         uint192 s = fixMin(sellAmount, maxTradeSize(sell));
 
         // {qSellTok}
         trade.sellAmount = s.shiftl_toUint(int8(sell.erc20Decimals()), FLOOR);
 
+        uint192 buyPrice = buy.strictPrice();
+
+        // buy.strictPrice() cannot be zero in BackingManager's usage of this function
+        // buy.strictPrice() could plausibly be zero in RevenueTrader's usage of this function
+        require(buyPrice > 0, "buy asset has zero price");
```

Comment:
So we should put this check in BackingManager itself, right?

Reply:
(proceeding assuming you mean `prepareTradeRecapitalize`)

And actually, `prepareTradeToCoverDeficit` requires the sellPrice to also be nonzero. I think this becomes an assert right after the `nextTradePair` call in `prepareTradeRecapitalize`.

Reply:
Mmm almost but not quite. Will need different constraints for the `prepareTradeSell` and `prepareTradeToCoverDeficit` cases.